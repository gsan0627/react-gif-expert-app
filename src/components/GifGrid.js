import React from "react";
import { GifGridItem } from "./GifGridItem";
import { useFetchGifs } from "../hooks/useFetchGifs";
import PropTypes from "prop-types";

export const GifGrid = ({ category }) => {
  const { data, loading } = useFetchGifs(category);

  return (
    <>
      <h3 className="animate__animated animate__bounce">{category}</h3>

      {loading && (
        <p className="animate__animated animate__flash">Loading...</p>
      )}

      <div className="card-grid">
        {data.map((image) => (
          <GifGridItem key={image.id} {...image} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
