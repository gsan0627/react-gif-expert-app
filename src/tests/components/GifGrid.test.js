import React from "react";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe("Component tests <GifGrid />", () => {
  const category = "Batman";
  test("the component should be displayed correctly", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("should show item when images are loaded with useFetchGifts", () => {
    const gifs = [
      { id: 1, title: "Batman", url: "https://localhost/anything.png" },
    ];
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("p").exists()).toBe(false);
    expect(wrapper.find("GifGridItem").length).toBe(gifs.length);
  });
});
