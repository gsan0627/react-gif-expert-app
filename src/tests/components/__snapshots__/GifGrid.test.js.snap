// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component tests <GifGrid /> should show item when images are loaded with useFetchGifts 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__bounce"
  >
    Batman
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id={1}
      key="1"
      title="Batman"
      url="https://localhost/anything.png"
    />
  </div>
</Fragment>
`;

exports[`Component tests <GifGrid /> the component should be displayed correctly 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__bounce"
  >
    Batman
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
